### PR TITLE
Upgrade to webpack 2

### PR DESCRIPTION
**Pull request by miltador**
This is my additions to #1063. Updated configs to webpack 2 and resolved conflicts.

### CODE REVIEW

**Comment by davezuko** [comment]
Took a look at this tonight; is `System.import` required now for webpack 2 instead of `require.ensure`? I'm not 100% up to speed with the changes just yet, so I'm curious to hear if this is a stylistic change on your part or if it's part of the upgrade path. I'll dig through their docs tonight and see if I can answer this myself.

**Comment by miltador** [comment]
@davezuko, about `System.import` you can read here: https://gist.github.com/sokra/27b24881210b56bbaff7#code-splitting-with-es6 . It is a new feature of ES6 and Webpack 2. However it's not directly related to upgrading webpack, old `require` approach will work too.

**Comment by shengnian** [comment]
upgrade to webpack 2, see: https://github.com/shengnian/webpack2-react-redux-starter,
but, has postcss config problem and test can't run, i cant's solve it, anyone can help,thanks.

**Comment by peterpme** [comment]
:wave: Anything I can do to help move this along?

**Comment by shengnian** [comment]
@peterpme thanks
first, clone this project [webpack2-react-redux](https://github.com/shengnian/webpack2-react-redux-starter)  and  into the project dirctory
`yarn install`
there have two problems:
**1:  PostCssConfig could not be loaded. Reproduce the error as follows:**
`yarn start`
will see the debug log:
```
app:bin:dev-server Server is now running at http://localhost:3000. +36ms
PostCSS Config could not be loaded. Please check your PostCSS Config.
PostCSS Config could not be loaded. Please check your PostCSS Config.
```

**2:  ERROR [karma]: TypeError: Cannot read property 'module' of null, when execute `yarn test`**
print debug log:

```
TypeError: Cannot read property 'module' of null
    at Resolver.<anonymous> (/data2/github_src/webpack2-react-redux-starter/node_modules/enhanced-resolve/lib/ParsePlugin.js:22:13)
    at Resolver.applyPluginsParallelBailResult1 (/data2/github_src/webpack2-react-redux-starter/node_modules/tapable/lib/Tapable.js:252:14)
    at beforeInnerCallback (/data2/github_src/webpack2-react-redux-starter/node_modules/enhanced-resolve/lib/Resolver.js:112:19)
    at loggingCallbackWrapper (/data2/github_src/webpack2-react-redux-starter/node_modules/enhanced-resolve/lib/createInnerCallback.js:31:19)
    at Resolver.applyPluginsAsyncSeriesBailResult1 (/data2/github_src/webpack2-react-redux-starter/node_modules/tapable/lib/Tapable.js:145:46)
    at Resolver.doResolve (/data2/github_src/webpack2-react-redux-starter/node_modules/enhanced-resolve/lib/Resolver.js:100:11)
    at Resolver.<anonymous> (/data2/github_src/webpack2-react-redux-starter/node_modules/enhanced-resolve/lib/UnsafeCachePlugin.js:36:12)
    at Resolver.applyPluginsParallelBailResult1 (/data2/github_src/webpack2-react-redux-starter/node_modules/tapable/lib/Tapable.js:252:14)
    at beforeInnerCallback (/data2/github_src/webpack2-react-redux-starter/node_modules/enhanced-resolve/lib/Resolver.js:112:19)
    at loggingCallbackWrapper (/data2/github_src/webpack2-react-redux-starter/node_modules/enhanced-resolve/lib/createInnerCallback.js:31:19)
    at Resolver.applyPluginsAsyncSeriesBailResult1 (/data2/github_src/webpack2-react-redux-starter/node_modules/tapable/lib/Tapable.js:145:46)
    at Resolver.doResolve (/data2/github_src/webpack2-react-redux-starter/node_modules/enhanced-resolve/lib/Resolver.js:100:11)
    at Resolver.resolve (/data2/github_src/webpack2-react-redux-starter/node_modules/enhanced-resolve/lib/Resolver.js:74:14)
    at /data2/github_src/webpack2-react-redux-starter/node_modules/webpack/lib/NormalModuleFactory.js:252:12
    at /data2/github_src/webpack2-react-redux-starter/node_modules/webpack/node_modules/async/dist/async.js:1008:9
    at eachOfArrayLike (/data2/github_src/webpack2-react-redux-starter/node_modules/webpack/node_modules/async/dist/async.js:941:9)
error Command failed with exit code 1.
```

**Comment by jonricaurte** [comment]
@davezuko Also keep in mind System.import is replaced by import()

https://github.com/webpack/webpack/issues/3098

**Comment by jonricaurte** [comment]
@shengnian  @peterpme  @davezuko  @miltador webpack 2.2.0 final was released yesterday

**Comment by shengnian** [comment]
@jonricaurte yes, i upgrade to webpack 2.2.0 and solved the two bugs, thanks.

**Comment by volodymyr-tsaryk** [comment]
What are the plans on this ?

**Comment by Mirodil** [comment]
any news on this?

**Comment by peterpme** [comment]
Hi everybody,

I put together a version for the universal-react-redux-starter kit that I can open source for you all. Then someone can abstract it and apply the changes themselves.

Any takers?

**Comment by Mirodil** [comment]
Hi @peterpme, 
I would like to try your universal-react-redux-starter. can you provide url?

**Comment by peterpme** [comment]
@Mirodil here's my PR for upgrading webpack 1 -> 2 for universal
https://github.com/janoist1/universal-react-redux-starter-kit/pull/25/files

File: build/webpack.config.js [hunk 1]
```diff
@@ -125,72 +150,48 @@ if (!__TEST__) {
 webpackConfig.module.loaders = [{
   test    : /\.(js|jsx)$/,
   exclude : /node_modules/,
-  loader  : 'babel',
+  loader  : 'babel-loader',
   query   : config.compiler_babel
 }, {
   test   : /\.json$/,
-  loader : 'json'
+  loader : 'json-loader'
```
Comment (vitkon) [inline build/webpack.config.js]: json loader is not required anymore

**Comment by shengnian** [comment]
webpack2 version: https://github.com/shengnian/webpack2-react-redux-starter.git

**Comment by Valentin-Seehausen** [comment]
+1 for support in this repo :)

**Comment by bmv437** [comment]
It looks like `System.import()` was replaced by `import()`
https://webpack.js.org/guides/code-splitting-import/#system-import-is-deprecated

**Comment by davezuko** [comment]
Moving to https://github.com/davezuko/react-redux-starter-kit/pull/1243.

Thanks for your work, I'll try to cherry-pick some of your fixes over. I have private project that just recently bumped webpack to 2.0, so I'm currently porting all of those changes over here.

**Pull request closed without merging**